pkg/chunk/cached_store.go

```diff
@@ -819,6 +819,9 @@ func NewCachedStore(storage object.ObjectStorage, config Config, reg prometheus.
 		if size == 0 || size > store.conf.BlockSize {
 			return
 		}
+		if store.bcache.exist(key) {
```

Comment:
we can use bcache.load()

Reply:
bcache.load will open cache file which is more costly. so i add bcache.exist function. if cache is scaned, only check block exist in keys map